met/src/libcode/vx_gis/dbf_file.cc

```diff
@@ -672,6 +662,382 @@ return;
 ////////////////////////////////////////////////////////////////////////
 
 
+   //
+   //  Code for class DbfFile
+   //
+
+
+////////////////////////////////////////////////////////////////////////
+
+
+DbfFile::DbfFile()
+
+{
+
+init_from_scratch();
+
+}
+
+
+////////////////////////////////////////////////////////////////////////
+
+
+DbfFile::~DbfFile()
+
+{
+
+close();
+
+}
+
+
+////////////////////////////////////////////////////////////////////////
+
+
+DbfFile::DbfFile(const DbfFile &)
+
+{
+
+mlog << Error << "\nDbfFile::DbfFile(const DbfFile &) -> "
+     << "should never be called!\n\n";
+
+exit ( 1 );
+
+}
+
+
+////////////////////////////////////////////////////////////////////////
+
+
+DbfFile & DbfFile::operator=(const DbfFile &)
+
+{
+
+mlog << Error << "\nDbfFile::operator=(const DbfFile &) -> "
+     << "should never be called!\n\n";
+
+exit ( 1 );
+
+return ( * this );
```

Comment:
It looks like this return will never be reached. I recommend removing it. Also, could consider using =delete instead of logging that the function should never be called to catch incorrect use earlier.

Reply:
@georgemccabe yes, agreed. We can switch using the =delete convention instead. In fact, I tested that change independently and it the compiler was happy with it. I'll write that up as a separate code cleanup issue so we can handle them in one swoop. I basically adapted an existing shapefile class over for the dbffile class. Unfortunately, we often get competing feedback in scans of the MET code. Sometime Fortify (or now SonarQube) complain that a line is unreachable. Other times, they complain when a function lacks a return statement.